src/core/ThemeProvider.tsx

```diff
@@ -0,0 +1,68 @@
+import { Spinner } from "@awsui/components-react";
+import * as React from "react";
+import { useEffect } from "react";
+import { useState } from "react";
+import { useToggle } from "../common/useToggle";
+
+type Theme = "light" | "dark";
+
+type ThemeContextReturn = [Theme, () => void];
+
+export const ThemeContext = React.createContext<ThemeContextReturn>([
+  "light",
+  () => {},
+]);
+
+interface Props {
+  className?: string;
+}
+
+async function getOSTheme(): Promise<Theme> {
+  // Getting OS Theme from Electron:
+  // https://www.electronjs.org/docs/tutorial/dark-mode
+  let timeout = 3; // seconds
```

Comment:
Why do we want to set up a timeout here?

Reply:
The main process (Electron app) and the renderer process (JS assets) runs in parallel. Only the main process has access to the OS to get the theme, so the renderer process triggers an event to fetch the dark mode, which is handled by the main process (src/index.ts line 45). The timeout (with 3 attempts) is to avoid race condition in case the assets load before the main process has had a chance to add the darkMode to the window (which happens async in the preload.ts)